pkg/cmd/extension/manager.go

```diff
@@ -172,7 +179,80 @@ func (m *Manager) InstallLocal(dir string) error {
 	return makeSymlink(dir, targetLink)
 }
 
-func (m *Manager) Install(cloneURL string, stdout, stderr io.Writer) error {
+type BinManifest struct {
+	Owner string
+	Name  string
+	Host  string
+	// TODO I may end up not using this; just thinking ahead to local installs
+	Path string
+}
+
+func (m *Manager) InstallBin(client *http.Client, repo ghrepo.Interface) error {
+	var r *release
+	r, err := fetchLatestRelease(client, repo)
+	if err != nil {
+		return err
+	}
+
+	suffix := m.platform()
+	var asset *releaseAsset
+	for _, a := range r.Assets {
+		if strings.HasSuffix(a.Name, suffix) {
+			asset = &a
+			break
+		}
+	}
+
+	if asset == nil {
+		return fmt.Errorf("%s unsupported for %s. Open an issue: `gh issue create -R%s/%s -t'Support %s'`",
+			repo.RepoName(),
+			suffix, repo.RepoOwner(), repo.RepoName(), suffix)
+	}
+
+	name := repo.RepoName()
+	targetDir := filepath.Join(m.installDir(), name)
+	// TODO clean this up if function errs?
```

Comment:
I think this is important to address, mainly due to the fact that without doing any cleanup the subsequent install attempts will likely fail due to the weird state of the install folder. Or we could make sure that subsequent install attempt just overwrite the install folder and its contents if it exists?

Reply:
It's easy enough to call `gh extension remove foo` to clean up so I'm not sure how urgent this is...I'm not opposed to doing it but might save it for a future improvement.